### PR TITLE
Add `deploy-account` command

### LINKED ISSUE #962: Support new `DeployAccount` transaction type (preferably as a CLI command)
https://starkware.notion.site/starkware/StarkNet-0-10-1-07b5c6d433464c61bf035037b8113b7b

### PR DESCRIPTION
**Pull request by kasperski95**
Allow sending "deploy account" transactions from the CLI.

Related #962. Docs will be added in another PR.

### CODE REVIEW

**Review by Arcticae — APPROVED** [review]
LGTM

File: protostar/commands/deploy_account_command.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                    f"Transaction hash: 0x{response.transaction_hash:064x}",
+                ]
+            )
+        )
+
+
+class AutoEstimateMaxFeeException(ProtostarException):
+    def __init__(self):
+        super().__init__(
+            "Protostar can't auto-estimate max fee for the DeployAccount transaction.\n"
```
Comment (mkaput) [inline protostar/commands/deploy_account_command.py:165]:
```suggestion
            "Protostar cannot auto estimate max fee for the DeployAccount transaction.\n"
```

File: protostar/commands/deploy_account_command.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        return response
+
+    def _map_typed_args_to_deploy_account_args(
+        self,
+        typed_args: DeployAccountCommandArgs,
+    ) -> DeployAccountArgs:
+        if typed_args.max_fee == "auto":
+            raise ProtostarException(
+                "Protostar can't auto-estimate max fee for the DeployAccount transaction."
+            )
```
Comment (mkaput) [inline protostar/commands/deploy_account_command.py:129]:
```suggestion
            raise AutoEstimateMaxFeeException()
```

File: website/docs/cli-reference.md [hunk 3]
```diff
@@ -123,6 +123,44 @@ An optional salt controlling where the contract will be deployed. The contract d
 Used by whitelisted users for deploying contracts in Alpha MainNet.
 #### `--wait-for-acceptance`
 Waits for transaction to be accepted on chain.
+### `deploy-account`
+Sends deploy-account transaction. The account contract must be already declared and prefunded.
```
Comment (mkaput) [inline website/docs/cli-reference.md:127]:
```suggestion
Deploys a new account on selected network. The account contract must be already declared and prefunded.
```

File: website/docs/cli-reference.md [hunk 4]
```diff
@@ -123,6 +123,44 @@ An optional salt controlling where the contract will be deployed. The contract d
...
+#### `--account-address STRING`
+Account address
```
Comment (mkaput) [inline website/docs/cli-reference.md:129]:
```suggestion
Account address.
```

**Review by mkaput — APPROVED** [review]

File: protostar/cli/protostar_arg_type.py [hunk 5]
```diff
@@ -29,7 +35,17 @@ def parse_felt_arg_type(arg: str) -> int:
     return output
 
 
+def parse_class_hash_arg_type(arg: str) -> int:
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply protostar/cli/protostar_arg_type.py:38]: This is "accidental duplication". IMO ClassHash is not an address.

[the remark this answers is not in the document]
Comment (Arcticae) [reply protostar/cli/protostar_arg_type.py:38]: Maybe we should deduplicate parsing even though naming is different

File: website/docs/cli-reference.md (hunk 3 above)
Comment (kasperski95) [reply website/docs/cli-reference.md:127]: I think the original version is more correct. Deployer deploys the account contract, not Protostar.

File: protostar/cli/protostar_arg_type.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (kasperski95) [reply protostar/cli/protostar_arg_type.py]: Done. Abstracted and reused parser.

File: website/docs/cli-reference.md (hunk 4 above)
Comment (kasperski95) [reply website/docs/cli-reference.md]: [update descriptions](https://github.com/software-mansion/protostar/pull/1083/commits/f5d59f69597d68d828ac69ea259e5c851c7b1502)

File: protostar/commands/deploy_account_command.py (hunk 2 above)
Comment (kasperski95) [reply protostar/commands/deploy_account_command.py]: [update descriptions](https://github.com/software-mansion/protostar/pull/1083/commits/f5d59f69597d68d828ac69ea259e5c851c7b1502)

File: protostar/commands/deploy_account_command.py (hunk 1 above)
Comment (kasperski95) [reply protostar/commands/deploy_account_command.py]: [update descriptions](https://github.com/software-mansion/protostar/pull/1083/commits/f5d59f69597d68d828ac69ea259e5c851c7b1502)

**Pull request merged**